### PR TITLE
chore(C): add linear search

### PR DESCRIPTION
**Pull request by arpitgoyal27**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

File: algorithms/C/searching/linearsearch.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// linear search
+#include <stdio.h>
+#define SIZE 100  //change according to question
```
Comment (UG-SEP) [inline algorithms/C/searching/linearsearch.c:3]: take size from the user so memory waste can be prevented and think if the size>100 then the program will terminate

File: algorithms/C/searching/linearsearch.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+int main()
+{
+   int data[ SIZE ], i, temp;
+   int searchQuery; //value to search in the array
+
+   //generate some data
+   for (i = 0; i < SIZE; ++i)
+   {
+      data[ i ] = 2 * i;
```
Comment (UG-SEP) [inline algorithms/C/searching/linearsearch.c:16]: what this you have to take array value from the user

File: algorithms/C/searching/linearsearch.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// linear search
+#include <stdio.h>
+#define SIZE 100  //change according to question
+
+//function prototype
+int linearSearch(int a[], int query, int size);
+
+int main()
+{
+   int data[ SIZE ], i, temp;
```
Comment (UG-SEP) [inline algorithms/C/searching/linearsearch.c:10]:
```suggestion
int *data,i,temp;
data=(int*)malloc(sizeof(int)*size);
```

File: algorithms/C/searching/linearsearch.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+   int searchQuery; //value to search in the array
```
Comment (UG-SEP) [inline algorithms/C/searching/linearsearch.c:11]: it would be better if you initialize it above

File: algorithms/C/searching/linearsearch.c [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+   printf("Enter integer to search for: ");
+   scanf("%d", &searchQuery);
+
+   //finding searchQuery in array data
+   temp = linearSearch(data, searchQuery, SIZE);
+
+   //condition for displaying result
+   if ( temp != -1)
+      printf("Found value in element %d.", temp);
```
Comment (UG-SEP) [inline algorithms/C/searching/linearsearch.c:27]:
```suggestion
      printf("Found value in element %d", temp);
```

File: algorithms/C/searching/linearsearch.c [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+   else
+      printf("Value not found.");
```
Comment (UG-SEP) [inline algorithms/C/searching/linearsearch.c:29]:
```suggestion
      printf("Value not found");
```

**Review by UG-SEP — CHANGES_REQUESTED** [review]
Hey @arpitgoyal27  do close the PR only  done the changes GitHub automatically add the changes in PR

**Comment by ming-tsai** [comment]
Hi @arpitgoyal27,
Do not close your pull request when update it

File: algorithms/C/searching/linearsearch.c [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+   printf("Enter integer to search for: ");
+   scanf("%d", &searchQuery);
+
+   //finding searchQuery in array data
+   temp = linearSearch(data, searchQuery, size);
+
+   //condition for displaying result
+   if ( temp != -1)
+      printf("Found value and the index is :%d", temp);
```
Comment (UG-SEP) [inline algorithms/C/searching/linearsearch.c:27]:
```suggestion
      printf("Value found at index no: %d", temp);
```
your sentence has not perfect sense

File: algorithms/C/searching/linearsearch.c [hunk 8]
```diff
@@ -0,0 +1,67 @@
+// linear search
+#include <stdio.h>
+#include <stdlib.h>
+//function prototype
+int linearSearch(int a[], int query, int size);
+
+int main()
+{
+   int *data,i,temp,searchQuery,size;
+   scanf("%d",&size);
```
Comment (UG-SEP) [inline algorithms/C/searching/linearsearch.c:10]: please add the UI here 
Hint: printf("Enter size: ");
So the user can understand that what to do

**Review by UG-SEP — CHANGES_REQUESTED** [review]
please do the change else everything looks  good

**Review by UG-SEP — APPROVED** [review]
Thanks for contributing @arpitgoyal27 
Hope you contribute more to this project

**Pull request merged**